### ISSUE TITLE
XiaListModeDataMask compile error 'invalid_argument'

### ISSUE THREAD

**Issue by ksmith0** [pixie16/paass#163]
The following errors were encountered attempting to build a clean dev branch at commit 9270d47 on r230 with g++ version 4.8.5 2010623 (Red Hat 4.8.5-11)

`BUILD_SHARED_LIBS=ON`, `BUILD_SKELETON=OFF`, `BUILD_UTKSCAN=OFF`
```
Scanning dependencies of target ScanObjects
[ 14%] Building CXX object Scan/ScanLib/source/CMakeFiles/ScanObjects.dir/XiaListModeDataMask.cpp.o
/home/ksmith/paass-temp/Scan/ScanLib/source/XiaListModeDataMask.cpp: In member function ‘DataProcessing::FIRMWARE XiaListModeDataMask::ConvertStringToFirmware(const string&)’:
/home/ksmith/paass-temp/Scan/ScanLib/source/XiaListModeDataMask.cpp:28:41: error: ‘invalid_argument’ was not declared in this scope
         throw invalid_argument(msg.str());
                                         ^
/home/ksmith/paass-temp/Scan/ScanLib/source/XiaListModeDataMask.cpp: In member function ‘std::pair<unsigned int, unsigned int> XiaListModeDataMask::GetCfdFractionalTimeMask() const’:
/home/ksmith/paass-temp/Scan/ScanLib/source/XiaListModeDataMask.cpp:39:68: error: ‘invalid_argument’ was not declared in this scope
                                        ("GetCfdFractionalTimeMask"));
                                                                    ^
/home/ksmith/paass-temp/Scan/ScanLib/source/XiaListModeDataMask.cpp: In member function ‘std::pair<unsigned int, unsigned int> XiaListModeDataMask::GetCfdForcedTriggerBitMask() const’:
/home/ksmith/paass-temp/Scan/ScanLib/source/XiaListModeDataMask.cpp:89:68: error: ‘invalid_argument’ was not declared in this scope
                                        ("GetCfdFractionalTimeMask"));
                                                                    ^
/home/ksmith/paass-temp/Scan/ScanLib/source/XiaListModeDataMask.cpp: In member function ‘std::pair<unsigned int, unsigned int> XiaListModeDataMask::GetCfdTriggerSourceMask() const’:
/home/ksmith/paass-temp/Scan/ScanLib/source/XiaListModeDataMask.cpp:129:68: error: ‘invalid_argument’ was not declared in this scope
                                        ("GetCfdFractionalTimeMask"));
                                                                    ^
/home/ksmith/paass-temp/Scan/ScanLib/source/XiaListModeDataMask.cpp: In member function ‘std::pair<unsigned int, unsigned int> XiaListModeDataMask::GetEventEnergyMask() const’:
/home/ksmith/paass-temp/Scan/ScanLib/source/XiaListModeDataMask.cpp:171:68: error: ‘invalid_argument’ was not declared in this scope
                                        ("GetCfdFractionalTimeMask"));
                                                                    ^
/home/ksmith/paass-temp/Scan/ScanLib/source/XiaListModeDataMask.cpp: In member function ‘std::pair<unsigned int, unsigned int> XiaListModeDataMask::GetTraceOutOfRangeFlagMask() const’:
/home/ksmith/paass-temp/Scan/ScanLib/source/XiaListModeDataMask.cpp:195:68: error: ‘invalid_argument’ was not declared in this scope
                                        ("GetCfdFractionalTimeMask"));
                                                                    ^
/home/ksmith/paass-temp/Scan/ScanLib/source/XiaListModeDataMask.cpp: In member function ‘std::pair<unsigned int, unsigned int> XiaListModeDataMask::GetTraceLengthMask() const’:
/home/ksmith/paass-temp/Scan/ScanLib/source/XiaListModeDataMask.cpp:222:68: error: ‘invalid_argument’ was not declared in this scope
                                        ("GetCfdFractionalTimeMask"));
                                                                    ^
/home/ksmith/paass-temp/Scan/ScanLib/source/XiaListModeDataMask.cpp: In member function ‘double XiaListModeDataMask::GetCfdSize() const’:
/home/ksmith/paass-temp/Scan/ScanLib/source/XiaListModeDataMask.cpp:251:68: error: ‘invalid_argument’ was not declared in this scope
                                        ("GetCfdFractionalTimeMask"));
                                                                    ^
make[2]: *** [Scan/ScanLib/source/CMakeFiles/ScanObjects.dir/XiaListModeDataMask.cpp.o] Error 1
make[1]: *** [Scan/ScanLib/source/CMakeFiles/ScanObjects.dir/all] Error 2
make: *** [all] Error 2
```

**Comment by pixiedummy** [comment]
This is currently being fixed in jenkins-test

**Comment by ksmith0** [comment]
Just so this is complete, XiaListModeDataDecoder needs stdexcept in the include list.

```c++
#include <stdexcept>
```

**Comment by spaulaus** [comment]
This was fixed with pull request #162
